compiler/pipes/final-check.cpp

```diff
@@ -509,6 +515,16 @@ VertexPtr FinalCheckPass::on_enter_vertex(VertexPtr vertex) {
     check_instanceof(vertex.try_as<op_instanceof>());
   }
 
+  if (vertex->type() == op_index) {
+    const auto index_vertex = vertex.as<op_index>();
+    check_indexing(index_vertex->array(), index_vertex->key());
```

Comment:
Don't you want to use .try_as? Instread of checking type and casting:
```
if (const auto v_index = vertex.try_as<op_index>()) {
  check_indexing(v_index->array(), v_index->key());
} 
// and below
```

Reply:
Done